CONTRIBUTING.md

```diff
@@ -0,0 +1,170 @@
+
+## Contributing Code
+Helping to improve the Reservations codebase is **great**. In general, we only accept pull requests for changes related to an open GitHub issue, so make sure you open one (and it's approved) before you start working. This also makes it easier to review and test your work. During active development there will be [milestones](https://github.com/YaleSTC/reservations/milestones) for upcoming versions; we'd prefer if you select open issues from those over issues in the [wish list](https://github.com/YaleSTC/reservations/milestones/Wish%20List). Otherwise, feel free to select any open issues, noting that we try to classify them by rough priority using the labels beginning with `pri:`.
+
+The following is an overview of how to work on Reservations:
+
+### Fork Reservations / Create a Branch
+Each developer working on Reservations should fork the core Reservations repository and clone it as a local repository. Remember to add the core repository as a remote:
+
+```
+git remote add upstream git@github.com:YaleSTC/Reservations.git
+```
+
+You can either work off the forked `master` branch or create a separate topic branch in your forked repository. Make sure you pull in the latest changes from `upstream` if you cloned a while ago (`git pull upstream master`)We generally recommend prefixing the branch name with the GitHub issue number and including a few descriptive words, i.e. `1234_brief_issue_description`.
+
+We generally maintain release branches for minor versions (e.g. `release-v3.4`) for patching purposes - if you're working on a version-specific bug make sure you branch off of the correct place.
+
+### Testing
+Make sure to update or add to the test suite where appropriate; patches and features will not be accepted without tests. We use [RSpec](http://rspec.info/) for general testing and [Capybara](http://jnicklas.github.io/capybara/) for integration testing. Make sure that the current test suite passes before submitting a PR by running `rake`.
+
+### Code / Style Conventions
+We use [Rubocop](http://batsov.com/rubocop/) as the arbiter of style for Reservations. You should run `rubocop -D` to ensure that all of your changes match the project conventions. We occasionally allow exceptions from code complexity violations; you should make any such request in the GitHub pull request comments.
+
+### Cleaning up your branch
+Before you submit your PR, you should clean up your commit history and resolve any merge conflicts with `master` (or your upstream branch). While we recommend making a bunch of small commits while you're working, it's important to maintain a readable and useful commit history in the main repository.
```

Comment:
We should be more explicit and say that we want them to squash commits when they're ready.